### PR TITLE
feat: Implement shadowing

### LINKED ISSUE #6: shadowing, spacing, grid

### PR DESCRIPTION
**Pull request by MahdiAbdi**
customize shadows and use them in a storybook.
Closes #6

### CODE REVIEW

File: libs/core/src/styles/constants.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Shadows } from '@mui/material';
+
+export const spacing = 4;
+
+export const shadows: Shadows = [
+  'none', // Flat
+  '0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)',
+  '0px 1px 2px 0px rgba(0,0,0,0.10)', // Raised
+  '0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)',
```
Comment (alisajadih) [inline libs/core/src/styles/constants.ts:9]: what are these?

Comment (MahdiAbdi) [reply libs/core/src/styles/constants.ts:9]: shadows from [MUI default theme](https://mui.com/material-ui/customization/default-theme/)

Comment (alisajadih) [reply libs/core/src/styles/constants.ts:9]: You don't need to copy them. 
shadows : [...theme.shadows , your shadows]

Comment (MahdiAbdi) [reply libs/core/src/styles/constants.ts:9]: The order makes difference. 
e.g: index 0 is default, 1 is custom, 2 is default, 3 is custom.

Comment (MahdiAbdi) [reply libs/core/src/styles/constants.ts:9]: unless I do something like this:
const shadows = theme.shadows;
shadows[1] = ...
shadows[24] =...

then export shadows;

File: libs/core/src/styles/Shadow.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { Grid, Paper } from '@mui/material';
+import { Meta } from '@storybook/react';
+
+export default {
```
Comment (alisajadih) [inline libs/core/src/styles/Shadow.stories.tsx:4]: why we need extra folder? 
Core/shadows?

File: libs/core/src/styles/constants.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { Shadows } from '@mui/material';
+
+export const spacing = 4;
```
Comment (alisajadih) [inline libs/core/src/styles/constants.ts:3]: you change the spacing. and buttons affected

File: libs/core/src/styles/constants.ts (hunk 1 above)
Comment (alisajadih) [reply libs/core/src/styles/constants.ts:9]: it's better. if default shadows change, you don't need to change them all or track them.

Comment (MahdiAbdi) [reply libs/core/src/styles/constants.ts:9]: then another question rises. what if the number of custom items increased? 10 or 20 custom shadows? then the above method won't be a good idea.

File: libs/core/src/styles/createTheme.tsx [hunk 4]
```diff
@@ -367,8 +368,9 @@ const defaultOptions: ThemeOptions = {
       },
     },
   },
-};
+});
 
 export function createTheme(options: ThemeOptions = {}, ...args: object[]): Theme {
-  return createMuiTheme(defaultOptions, options, ...args);
+  const defaultTheme = createMuiTheme();
+  return createMuiTheme(createDefaultOptions(defaultTheme), options, ...args);
```
Comment (mdeljavan) [inline libs/core/src/styles/createTheme.tsx:375]:
```suggestion
  const theme = createMuiTheme(createDefaultOptions(defaultTheme), options, ...args);
  theme.shadows = createShadows(theme.shadows);
  return theme
```

File: libs/core/src/styles/constants.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { Shadows } from '@mui/material';
+
+export const createShadows = (defaultShadows: Shadows): Shadows =>
```
Comment (alisajadih) [inline libs/core/src/styles/constants.ts:3]: filename

File: libs/core/src/styles/createTheme.tsx [hunk 6]
```diff
@@ -367,8 +368,9 @@ const defaultOptions: ThemeOptions = {
       },
     },
   },
-};
+});
 
 export function createTheme(options: ThemeOptions = {}, ...args: object[]): Theme {
-  return createMuiTheme(defaultOptions, options, ...args);
+  const defaultTheme = createMuiTheme();
```
Comment (alisajadih) [inline libs/core/src/styles/createTheme.tsx:374]: defaultMuiTheme

File: libs/core/src/styles/constants.ts (hunk 5 above)
Comment (mdeljavan) [inline libs/core/src/styles/constants.ts:3]: IMO this function is util

**Review by mdeljavan — CHANGES_REQUESTED** [review]

File: libs/core/src/styles/constants.ts (hunk 5 above)
Comment (MahdiAbdi) [reply libs/core/src/styles/constants.ts]: what do you think about the new change?
https://github.com/frontzen/design-system/pull/20/commits/bd4ddcdd5eb987f5ec3a0fbeb4b2b90727e397a3

File: libs/core/src/styles/createTheme.tsx (hunk 4 above)
Comment (MahdiAbdi) [reply libs/core/src/styles/createTheme.tsx]: I don't like this.
what if we decided to change some other styles and they needed the defaultMuiTheme.
adding them to createTheme function doesn't seem like a great idea.

Comment (MahdiAbdi) [reply libs/core/src/styles/createTheme.tsx]: fixed it.

**Review by mdeljavan — APPROVED** [review]

**Review by alisajadih — DISMISSED** [review]

**Review by mdeljavan — APPROVED** [review]

**Pull request merged**